src/foam/java/refinements.js

```diff
@@ -112,6 +117,7 @@ foam.CLASS({
       var constantize = foam.String.constantize(this.name);
       var isSet       = this.name + 'IsSet_';
       var factoryName = capitalized + 'Factory_';
+      var assertName  = capitalized + 'Assert_';
```

Comment:
I think the name should be more friendly in this case. So getFoo(), setFoo(), and assertFoo(). Then below, instead of checking if javaSetter is set or not, we always make the assertFoo() call. This is friendlier for sub-classing and for those marking their models as abstract and putting larger code-blocks in a Java file. In the case where the implementation is empty, Java will just eliminate the call anyway.

Reply:
Done